Tests/PostgreSQLTests/PostgreSQLConnectionTests.swift

```diff
@@ -236,13 +236,26 @@ class PostgreSQLConnectionTests: XCTestCase {
 
     func testURLParsing() throws {
         let databaseURL = "postgres://username:password@localhost:5432/database"
-        let config = try PostgreSQLDatabaseConfig(url: databaseURL)!
+        let config = PostgreSQLDatabaseConfig(url: databaseURL)!
         XCTAssertEqual("\(config.serverAddress)", "ServerAddress(storage: PostgreSQL.PostgreSQLConnection.ServerAddress.Storage.tcp(hostname: \"localhost\", port: 5432))")
         XCTAssertEqual(config.username, "username")
         XCTAssertEqual(config.password, "password")
         XCTAssertEqual(config.database, "database")
     }
 
+    func testURLSocketParsing() throws {
+        let databaseURL = "postgres://username:password@%2Fprivate%2Ftmp:5433/database"
+        let config = PostgreSQLDatabaseConfig(url: databaseURL)!
+        #if swift(>=4.1.50)  // compiler 4.2
```

Comment:
nit: While you are at it, could you also augment `testURLParsing` in the same way? (Personally, I'd prefer to actually check the components instead of comparing string representations, but given that `testURLParsing` already compares string representations, I won't object.)

Reply:
I made the change to check the storage on both tests. That seems more robust.